pkg/virtualization/core/storage/disk/virtualharddisksettingdata.go

```diff
@@ -74,3 +91,75 @@ func GetVirtualHardDiskSettingData(whost *host.WmiHost, path string,
 
 	return
 }
+
+func GetVirtualHardDiskSettingDataFromXml(whost *host.WmiHost, xmlInstance string) (size uint64,
+	blockSize uint32,
+	lSectorSize uint32,
+	pSectorSize uint32,
+	format uint16,
+	err error) {
+	/*vhdTemp, err := GetDefaultVirtualHardDiskSettingData(whost)
+	if err != nil {
+		return nil, err
+	}
+	defer vhdTemp.Close()
+
+	// clone vhdsetting and populated the cloned instance
+	vhdWmi, err := NewVirtualHardDiskSettingData(vhdTemp.WmiInstance)
+	if err != nil {
+		return nil, err
+	}
+
+	vhdWmiClone, err := vhdWmi.Clone()
+	if err != nil {
+		return nil, err
+	}
+
+	vhdsetting, err = NewVirtualHardDiskSettingData(vhdWmiClone)
+	if err != nil {
+		return nil, err
+	}*/
+
+	log.Printf("Decoding WMI response [%s]\n", xmlInstance)
+	var tempvar uint64
+	var diskData INSTANCE
+	err = xml.Unmarshal([]byte(xmlInstance), &diskData)
+	if err != nil {
+		return
+	}
+
+	for _, property := range diskData.PROPERTY {
+		if property.NAME == "MaxInternalSize" {
+			size, err = strconv.ParseUint(property.VALUE, 10, 64)
+			if err != nil {
+				return
+			}
+		} else if property.NAME == "BlockSize" {
+			tempvar, err = strconv.ParseUint(property.VALUE, 10, 32)
+			if err != nil {
+				return
+			}
+			blockSize = uint32(tempvar)
+		} else if property.NAME == "LogicalSectorSize" {
+			tempvar, err = strconv.ParseUint(property.VALUE, 10, 32)
+			if err != nil {
+				return
+			}
+			lSectorSize = uint32(tempvar)
+		} else if property.NAME == "PhysicalSectorSize" {
+			tempvar, err = strconv.ParseUint(property.VALUE, 10, 32)
+			if err != nil {
+				return
+			}
+			pSectorSize = uint32(tempvar)
+		} else if property.NAME == "Format" {
```

Comment:
I guess it would be my preference and/or clean look if you are using 'switch' instead of if-else chain.

switch  property.NAME {
 case "MaxInternalSize":
...
 case "BlockSize":
....
}

Reply:
Fixed